Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionContextCocoa.mm

```diff
@@ -1186,9 +1214,67 @@ static _WKWebExtensionContextError toAPI(WebExtensionContext::Error error)
     m_backgroundWebView = nil;
 }
 
+WebExtensionContext::EventListenterTypeCountedSet WebExtensionContext::loadBackgroundPageListenersFromStorage()
```

Comment:
Instead of having this return a copy of the set, it would be better to just have it update `m_backgroundPageListeners`. You should call `m_backgroundPageListeners.clear()` at the start to make sure it does not duplicate data if it is called multiple times.